packages/loader/container-loader/src/loader.ts

```diff
@@ -363,13 +379,19 @@ export class Loader extends EventEmitter implements ILoader {
             return Promise.reject(new Error(`Invalid URL ${resolvedAsFluid.url}`));
         }
 
+        if (pendingLocalState !== undefined && resolvedAsFluid.url !== pendingLocalState.url) {
+            const message = `URL ${resolvedAsFluid.url} does not match pending state URL ${pendingLocalState.url}`;
```

Comment:
throw new Error()... is what we are trying to standardize across code base for async functions, I believe